lib/dalli/client.rb

```diff
@@ -359,12 +359,13 @@ def ring
 
     # Chokepoint method for instrumentation
     def perform(*all_args)
-      return yield if block_given?
-      op, key, *args = *all_args
-
-      key = key.to_s
-      key = validate_key(key)
       begin
+        return yield if block_given?
```

Comment:
This needs to be moved under the begin so that perform will retry with and without a block.